internal/core/database_api/athena/driver/api/notify_appsync.go

```
@@ -0,0 +1,136 @@
+       workflowId
+     }
+}
+`
+)
+
+type GraphQlQuery struct {
+	Query string `json:"query"` // when we marshal, this will escape the mutation JSON as required by graphQL
+}
+
+type GraphQlResponse struct {
+	Data   interface{}   `json:"data"`
+	Errors []interface{} `json:"errors"`
+}
+
+func (API) NotifyAppSync(input *models.NotifyAppSyncInput) (*models.NotifyAppSyncOutput, error) {
+	var output models.NotifyAppSyncOutput
+
+	var err error
+	defer func() {
+		if err != nil {
+			err = apiError(err) // lambda failed
+		}
+
+		// allows tracing queries
+		zap.L().Info("NotifyAppSync",
+			zap.String("userData", input.UserData),
+			zap.String("queryId", input.QueryID),
+			zap.String("workflowID", input.WorkflowID),
+			zap.Error(err))
+	}()
+
+	// make sigv4 https request to appsync endpoint notifying query is complete, sending  userData, queryId and workflowId
+	var httpClient http.Client
+	signer := v4.NewSigner(awsSession.Config.Credentials)
+
+	mutation := &GraphQlQuery{
+		Query: fmt.Sprintf(mutationTemplate, input.UserData, input.QueryID, input.WorkflowID),
+	}
+	jsonMessage, err := jsoniter.Marshal(mutation)
+	if err != nil {
+		err = errors.Wrapf(err, "json marshal failed for: %#v", input)
+		return &output, err
+	}
+
+	body := bytes.NewReader(jsonMessage) // JSON envelope for graphQL
+
+	req, err := http.NewRequest("POST", envConfig.GraphqlEndpoint, body)
+	if err != nil {
+		err = errors.Wrapf(err, "new htttp request failed for: %#v", input)
+		return &output, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	_, err = signer.Sign(req, body, "appsync", *awsSession.Config.Region, time.Now().UTC())
+	if err != nil {
+		err = errors.Wrapf(err, "failed to v4 sign %#v", input)
+		return &output, err
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to POST %#v", req)
+		return &output, err
+	}
+	defer resp.Body.Close()
+
+	output.StatusCode = resp.StatusCode
+
+	respBody, _ := ioutil.ReadAll(resp.Body)
```

Why are we ignoring the error? We should be checking if there is an error returned and return it to the caller.

---

The line after does the error check, it not necessary.

---

Added a clarifying comment